### PR TITLE
Support v2306

### PR DESCRIPTION
**Pull request by YT-Advanced**
Add support for v2306 (WSA switch back to EXT4 Image)

- [x] #615 
- [x] Have tested the modifications

### CODE REVIEW

**Review by Almighty188 — COMMENTED** [review]
Works fine in building WSA v2306.40000.1.0_x64

**Review by MustardChef — COMMENTED** [review]
Can confirm that it works.

**Comment by theAeon** [comment]
Seems to break Amazon Appstore on upgrade.

Whether that's a bug or a feature is up in the air.

**Comment by Almighty188** [comment]
> Seems to break Amazon Appstore on upgrade.
> 
> Whether that's a bug or a feature is up in the air.

Mines is not technically a upgrade as Play Store and Magisk among other apps puts the Starting WSA is starting up for atleast 5 minutes and then the window disappears and the WSA service in the background also terminates until I uninstall WSA first and then install again as mentioned here:
https://github.com/LSPosed/MagiskOnWSALocal/discussions/622

I did not test Amazon Store in the 2305 versions but just tried it now and it works:
image
Amazon Appstore is version release-60.17.1.0.208834.0_426010

**Comment by YT-Advanced** [comment]
@Almighty188 @theAeon Because the all the image (include userdata.vhdx) switch from EROFS to EXT4 so you must reinstall WSA

**Comment by Almighty188** [comment]
@YT-Advanced - That makes perfect sense but I had the problem with the last 2 2305 versions as well so it could be a app that is causing the corruption as well.  Is there a way to replace the userdata.vhdx without reinstalling?

**Comment by theAeon** [comment]
image

**Comment by YT-Advanced** [comment]
@Almighty188 I think no way

**Comment by Almighty188** [comment]
@YT-Advanced - I just built new WSA 2306.40000.3.0 and I noticed in the output folder, where the Run.bat file is, there is a supplied userdata.vhdx file so it would seem one can just replace the %LOCALAPPDATA%\Packages\MicrosoftCorporationII.WindowsSubsystemForAndroid_8wekyb3d8bbwe\LocalCache\userdata.vhdx file from 
output\WSA_2306.40000.3.0_x64_Release-Nightly-with-magisk-26.1(26100)-stable-MindTheGapps-13.0\userdata.vhdx 

Haven't tried it though since the upgrade to 2306.40000.3.0 from 2306.40000.2.0 works fine.

**Review by skycat2216 — COMMENTED** [review]
The mount process has completed sucessfully

COMMAND_LINE=--arch x64 --release-type WIS --root-sol kernelsu --gapps-brand MindTheGapps
INFO: Architecture: x64
INFO: Release Type: WIS
INFO: Magisk Version: stable
INFO: GApps Brand: MindTheGapps
INFO: GApps Variant: pico
INFO: Root Solution: kernelsu
Build: RELEASE_TYPE=Insider Slow

**Comment by mio-19** [comment]
Does WSA 2306.x require manual downloading? Insider branches on the script gave me 2305

**Comment by Almighty188** [comment]
> Does WSA 2306.x require manual downloading? Insider branches on the script gave me 2305
> 
> Got 2306.40000.1 [#615 (comment)](https://github.com/LSPosed/MagiskOnWSALocal/issues/615#issuecomment-1625863672)
> 
> Where can I find links of .2/.3/.4?

Look in the Discussions area where I posted about each version being released.
https://github.com/LSPosed/MagiskOnWSALocal/discussions

File: scripts/build.sh
Comment (straker135) [inline scripts/build.sh:1]: Worked perfectly thanks very much. Prior to this I was getting the efocs error so no 'Output' folder was generated. The script updates fixed the problem completely.

Fabulous work

**Comment by yige-yigeren** [comment]
Can confirm that it works.
I create a brench to test it, it now instand main brench (no work).
https://github.com/yige-yigeren/MagiskOnWSAOnlineBuild/tree/switch-source-temporarily

**Comment by YT-Advanced** [comment]
Look like the repos owner stopped their work 😑

**Comment by MustardChef** [comment]
> Look like the repos owner stopped their work 😑

Don't think so. Last time, it was the same thing with EROFS.

**Review by yige-yigeren — COMMENTED** [review]
可以工作

**Comment by Almighty188** [comment]
> > Look like the repos owner stopped their work 😑
> 
> Don't think so. Last time, it was the same thing with EROFS. Let's wait. 😊

People don' realize this is a volunteer effort out of their own goodwill and no one is under any obligation to provide anything.

**Comment by yige-yigeren** [comment]
> Look like the repos owner stopped their work 😑

Don't be discouraged, the main content maintainers are college students, and it's the summer holidays, so maybe they're busy looking for jobs, going home or something else.
It should get better after a while.

不要灰心，主要的内容维护者是大学生，现在是暑假期间，也许他们正忙着找工作，回家或者其他事情。
过段时间应该就会好起来的

**Review by zh2209645 — COMMENTED** [review]
Win11 22H2 22621.2134
WSA version 2306.40000.4.0
Builded successfully in WSL2

**Review by yige-yigeren — COMMENTED** [review]
it seem that also work on 2307

My online build project show that is OK

https://github.com/yige-yigeren/MagiskOnWSAOnlineBuild/releases/tag/WSA_2307.40000.5.0

**Comment by YT-Advanced** [comment]
> it seem that also work on 2307.40000.5.0 似乎在最新版的2307.40000.5.0上也正常工作
> 
> My online build project show that is OK 我的一个在线构建项目成功部署了
> 
> https://github.com/yige-yigeren/MagiskOnWSAOnlineBuild/releases/tag/WSA_2307.40000.5.0

@yige-yigeren You should take a look in my repos:
https://github.com/YT-Advanced/WSA-Script

**Comment by yige-yigeren** [comment]
> > 似乎也适用于 2307.40000.5.0 似乎在最新版的 2307.40000.5.0 上也正常工作
> > 我的在线构建项目显示没问题 我的一个在线构建项目成功部署了
> > https://github.com/yige-yigeren/MagiskOnWSAOnlineBuild/releases/tag/WSA_2307.40000.5.0
> 
> 你应该看看我的存储库：https://github.com/YT-Advanced/WSA-Script

WOW, it looks like we both had similar ideas, I didn't find a similar project at the time, and in fact I'm not very good with GitHub Action, so I just wrote the simple faux-user action MagiskOnWSALocal logic and posted it for people who needed it but couldn't build it themselves.
I'm actually using this fork of yours for my repository as well, maybe if they don't update again I'll switch the base update repository from default to your project >_<
You can see the BASE_REPOSITORY:marked above the latest few releases is this one @. @
https://github.com/yige-yigeren/MagiskOnWSAOnlineBuild/releases

wow，看起来我们俩都有相似的想法，我当时并没有找到类似的项目，实际上我不是很会用GitHub Action，所以只写了简易的仿用户操作MagiskOnWSALocal逻辑，并且发布给有需要但自己又不会构建的人。
实际上我的仓库也正在使用你的这个分叉，也许他们再不更新我会将基础更新仓库从默认切换到你的项目>_<
你可以看到最新几个发布上面标记的BASE_REPOSITORY:就是这个@.@
https://github.com/yige-yigeren/MagiskOnWSAOnlineBuild/releases

**Comment by yige-yigeren** [comment]
> > it seem that also work on 2307.40000.5.0 似乎在最新版的2307.40000.5.0上也正常工作
> > My online build project show that is OK 我的一个在线构建项目成功部署了
> > https://github.com/yige-yigeren/MagiskOnWSAOnlineBuild/releases/tag/WSA_2307.40000.5.0
> 
> @yige-yigeren You should take a look in my repos: https://github.com/YT-Advanced/WSA-Script

I haven't found a way to write i18n in a small paragraphs in release yet, so some of it is only in Chinese QwQ

**Comment by hartmark** [comment]
I had same problem as #661 and checking out this branch fixes the issue.

**Comment by Tokkioo** [comment]
New at linux, how can I open the build.sh to edit the lines?

**Comment by zh2209645** [comment]
> New at linux, how can I open the build.sh to edit the lines?

Use any editor you like. Make sure you have installed it on your system.

**Comment by Anivie** [comment]
Thank you, I spent almost a night to fix this error, still I saw your pr

**Comment by yige-yigeren** [comment]
> New at linux, how can I open the build.sh to edit the lines?

If you not good at linux, may you can try wsl2, just use mnt - C/D/E/F to run script in windows

**Comment by shubhamsinghshubham777** [comment]
Checking if someone is active enough to review and merge at least this PR only, please?

**Pull request closed without merging**